scipy/optimize/_highs/setup.py

```diff
@@ -0,0 +1,133 @@
+
+import sys
+import pathlib
+from datetime import datetime
+
+
+def _get_sources(CMakeLists, start_token, end_token):
+    # Read in sources from CMakeLists.txt
+    CMakeLists = pathlib.Path(__file__).parent / CMakeLists
+    with open(CMakeLists, 'r') as f:
+        s = f.read()
+
+        # Find block where sources are listed
+        start_idx = s.find(start_token) + len(start_token)
+        end_idx = s[start_idx:].find(end_token) + len(s[:start_idx])
+        sources = s[start_idx:end_idx].split('\n')
+        sources = [s.strip() for s in sources if s[0] != '#']
+
+    # Make relative to setup.py
+    sources = [str(pathlib.Path('src/' + s)) for s in sources]
+    return sources
+
+
+# Grab some more info about HiGHS from root CMakeLists
+def _get_version(CMakeLists, start_token, end_token=')'):
+    CMakeLists = pathlib.Path(__file__).parent / CMakeLists
+    with open(CMakeLists, 'r') as f:
+        s = f.read()
+        start_idx = s.find(start_token) + len(start_token) + 1
+        end_idx = s[start_idx:].find(end_token) + len(s[:start_idx])
+    return s[start_idx:end_idx].strip()
+
+
+def configuration(parent_package='', top_path=None):
+
+    from numpy.distutils.misc_util import Configuration
+    config = Configuration('_highs', parent_package, top_path)
+
+    # HiGHS info
+    HIGHS_VERSION_MAJOR = _get_version('CMakeLists.txt', 'HIGHS_VERSION_MAJOR')
+    HIGHS_VERSION_MINOR = _get_version('CMakeLists.txt', 'HIGHS_VERSION_MINOR')
+    HIGHS_VERSION_PATCH = _get_version('CMakeLists.txt', 'HIGHS_VERSION_PATCH')
+    GITHASH = 'n/a'
+    HIGHS_DIR = str(pathlib.Path(__file__).parent.resolve())
+
+    # Here are the pound defines that HConfig.h would usually provide;
+    # We provide an empty HConfig.h file and do the defs and undefs
+    # here:
+    TODAY_DATE = datetime.today().strftime('%Y-%m-%d')
+    # ('OPENMP', None), ?
+    DEFINE_MACROS = [
+        ('CMAKE_BUILD_TYPE', '"Release"'),
+        ('HiGHSRELEASE', None),
+        ('IPX_ON', 'ON'),
+        ('HIGHS_GITHASH', '"%s"' % GITHASH),
+        ('HIGHS_COMPILATION_DATE', '"' + TODAY_DATE + '"'),
+        ('HIGHS_VERSION_MAJOR', HIGHS_VERSION_MAJOR),
+        ('HIGHS_VERSION_MINOR', HIGHS_VERSION_MINOR),
+        ('HIGHS_VERSION_PATCH', HIGHS_VERSION_PATCH),
+        ('HIGHS_DIR', '"' + HIGHS_DIR + '"'),
+    ]
+    UNDEF_MACROS = [
+        'OPENMP',  # unconditionally disable openmp
+        'EXT_PRESOLVE',
+        'SCIP_DEV',
+        'HiGHSDEV',
+        'OSI_FOUND',
+    ]
+
+    # HiGHS written using C++11 standard, we use C++14 for std::make_unique;
```

Comment:
Checked this is not blocked by MSVC; it's available in MSVC 2013 while our minimum requirement is MSVC 2015

Reply:
Apologies, I did not realize c++14 was not already being used in scipy.  I've removed calls to `std::make_unique` and is now c++11 compliant.

I've had performance issues with c++14 vs c++11 before, perhaps this will help with passing the remaining CI timeout issue